gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -33,20 +34,87 @@
  * @see <a href="https://cloud.google.com/dns/zones/">Google Cloud DNS managed zone
  * documentation</a>
  */
-public class Zone implements Serializable {
+public class Zone extends ZoneInfo {
 
-  // TODO(mderka) Zone and zoneInfo to be merged. Opened issue #605.
+  private static final long serialVersionUID = 564454483894599281L;
+  private transient Dns dns;
```

Comment:
You need to keep the options, consider it in equals/hashCode and implement readObject for assign a value to dns based on the options. See [Bucket](https://github.com/GoogleCloudPlatform/gcloud-java/blob/cdec697470a3e55066936d7099a0f066ae5f3005/gcloud-java-storage/src/main/java/com/google/gcloud/storage/Bucket.java#L406) as an example.